### PR TITLE
Progress bar

### PR DESCRIPTION
**Pull request by fisyher**
This branch contains progress bar implementation, including live performance update, display of progress bar in result screen, saving of progress bar records and display of progress bar during Song Selection.

### CODE REVIEW

File: DTXMania/Code/App/CDTXMania.cs [hunk 1]
```diff
@@ -1420,10 +1557,13 @@ protected override void Draw(GameTime gameTime)
                                             nRank = CScoreIni.tCalculateOverallRankValue(cPerfEntry_Drums, cPerfEntry_Guitar, cPerfEntry_Bass);
                                             strInstrument = " G+B";
                                         }
-                                        // Guitar only played
-                                        strPerfSkill = String.Format(" {0:F2}", cPerfEntry_Guitar.dbPerformanceSkill);
-                                        nRank = (CDTXMania.ConfigIni.nSkillMode == 0) ? CScoreIni.tCalculateRankOld(cPerfEntry_Guitar) : CScoreIni.tCalculateRank(0, cPerfEntry_Guitar.dbPerformanceSkill);
-                                        strInstrument = " Guitar";
+                                        else
```
Comment (limyz) [inline DTXMania/Code/App/CDTXMania.cs:1560]: good catch

File: DTXMania/Code/Score,Song/CSongManager.cs [hunk 2]
```diff
@@ -1814,6 +1822,14 @@ public void tReadScoreIniAndSetScoreInformation( string strScoreIniファイル
 					score.SongInformation.HighSkill[ n楽器番号 ] = ini.stSection[ n ].dbPerformanceSkill;
                     score.SongInformation.HighSongSkill[ n楽器番号 ] = ini.stSection[ n ].dbGameSkill;
 					score.SongInformation.FullCombo[ n楽器番号 ] = ini.stSection[ n ].bIsFullCombo | ini.stSection[ n楽器番号 * 2 ].bIsFullCombo;
+					//New for Progress
+					score.SongInformation.progress[n楽器番号] = ini.stSection[n].strProgress;					
+					if(score.SongInformation.progress[n楽器番号] == "")
+                    {
```
Comment (limyz) [inline DTXMania/Code/Score,Song/CSongManager.cs:1828]: indentation

File: DTXMania/Code/Stage/07.Performance/GuitarScreen/CActPerfGuitarBonus.cs [hunk 3]
```diff
@@ -64,7 +64,7 @@ public override unsafe int OnUpdateAndDraw()
 
                 if (CDTXMania.ConfigIni.bShowScore)
                 {
-                    drawBonusScoreAnimation(EInstrumentPart.GUITAR, new Point(303, 45));
+                    drawBonusScoreAnimation(EInstrumentPart.GUITAR, new Point(333, 45));
```
Comment (limyz) [inline DTXMania/Code/Stage/07.Performance/GuitarScreen/CActPerfGuitarBonus.cs:67]: omg never ending hard coded values

File: DTXMania/Code/Stage/08.Result/CStageResult.cs [hunk 4]
```diff
@@ -50,6 +53,7 @@ public CStageResult()
 			base.listChildActivities.Add( this.actParameterPanel = new CActResultParameterPanel() );
 			base.listChildActivities.Add( this.actRank = new CActResultRank() );
 			base.listChildActivities.Add( this.actSongBar = new CActResultSongBar() );
+			//base.listChildActivities.Add( this.actProgressBar = new CActPerfProgressBar(true) );
```
Comment (limyz) [inline DTXMania/Code/Stage/08.Result/CStageResult.cs:56]: we need to start looking into removing unreferenced functions

**Review by limyz — APPROVED** [review]
awesome work

File: DTXMania/Code/Stage/07.Performance/GuitarScreen/CActPerfGuitarBonus.cs (hunk 3 above)
Comment (fisyher) [reply DTXMania/Code/Stage/07.Performance/GuitarScreen/CActPerfGuitarBonus.cs:67]: Drawing positions of textures are all hard-coded in this project. Will need to major overhaul and refactoring if we want to make it configurable or auto-scale with screen resolution.

**Pull request merged**